servers/src/grin/server.rs

```diff
@@ -259,11 +263,14 @@ impl Server {
 		let p2p_inner = p2p_server.clone();
 		let p2p_thread = thread::Builder::new()
 			.name("p2p-server".to_string())
-			.spawn(move || {
-				if let Err(e) = p2p_inner.listen() {
-					error!("P2P server failed with erorr: {:?}", e);
-				}
-			})?;
+			.spawn(move || p2p_inner.listen())?;
```

Comment:
why not handle `Err` case?

Reply:
I wasn't sure about that either when reviewing the code. Think it might be because the call is wrapped inside the `if`, making it a little more complicated to get the `JoinHandle` for error handling below.

Can work on a solution for getting the error handle out, maybe defining a `mut Option` var outside the `if`.

Reply:
Added an `i2p_thread` handle in server, and handle errors in shutdown. Thanks for the catch.